src/sparseml/pytorch/datasets/ffcv_dataset.py

```diff
@@ -0,0 +1,299 @@
+            and it's corresponding FFCV fields have been defined.
+        """
+
+        with self.switch_off_dataset_arg("transform"), self.switch_off_dataset_arg(
+            "target_transform"
+        ):
+            self._write_if_not_written_already(
+                write_path=write_path,
+                num_workers=num_workers,
+            )
+
+        LOGGER.info("Dataset successfully written to disk.")
+
+    @timeit
+    def _write_if_not_written_already(
+        self,
+        write_path: str,
+        num_workers: int = 16,
+    ):
+        dataset_path = Path(write_path)
+
+        # Skip if dataset is already written
+        if dataset_path.exists():
+            LOGGER.info("Dataset already written to disk. Skipping _write.")
+            return
+
+        # Make parents if they don't exist
+        dataset_path.parent.mkdir(parents=True, exist_ok=True)
+
+        # Initialize the writer
+        writer = DatasetWriter(
+            fname=write_path,
+            fields=self.ffcv_fields,
+            num_workers=num_workers,
+        )
+
+        LOGGER.info(f"Writing dataset to {write_path}")
+
+        # self is the dataset
+        writer.from_indexed_dataset(self)
+
+    @contextmanager
+    def switch_off_dataset_arg(self, arg):
+        """
+        Context manager to perform an operation with a dataset argument turned
+        to None
+        """
+
+        old_arg_value = None
+
+        try:
+            # Set dataset attribute to None
+            if hasattr(self, arg):
+                old_arg_value = getattr(self, arg)
+                setattr(self, arg, None)
+                LOGGER.info(f"Dataset attribute {arg} set to {None}")
+            # Give back control
+            yield
+
+        finally:
+            # Restore dataset attribute to its original state
+            if old_arg_value is not None and hasattr(self, arg):
+                setattr(self, arg, old_arg_value)
+                LOGGER.info(f"Dataset attribute {arg} restored to old state")
+
+    def get_ffcv_loader(
+        self,
+        write_path: str,
+        num_workers: int = 16,
+        batch_size: int = 32,
+        distributed: bool = False,
+        in_memory: bool = False,
+        device: Union[str, int] = default_device(),
+    ):
+        """
+        Initialize the data loader.
+
+        :param write_path: The path to write the dataset to.
+        :param num_workers: Number of workers to use for the data loader.
+        :param batch_size: Batch size for the data loader.
+        :param distributed: bool Whether to use distributed data loading.
+        :param in_memory: bool Does the dataset fit in memory.
+        :param device: str or int The device to use for the data loader.
+        """
+
+        if not ffcv:
+            ffcv_url = "https://github.com/libffcv/ffcv#readme"
+            raise ImportError(
+                "FFCV is not installed. Please install using instructions from "
+                f"{ffcv_url}"
```

Comment:
let's not point to the ffcv url for now, as we may add our own docs. can just say FFCV not installed